### PR TITLE
Added summary algorithm

### PR DESCRIPTION
**Pull request by bartvanb**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,10 @@ echo:
 	@echo ""
 
 chisq:
-	make docker IMAGE=starter/vtg.chisq
+	make docker PKG_NAME=vtg.chisq
+
+summary:
+	make docker PKG_NAME=vtg.summary
```
Comment (frankcorneliusmartin) [inline Makefile:16]: You should add the `PKG_NAME=vtg.summary` here as well.

File: vtg.preprocessing/R/subset_data.R [hunk 2]
```diff
@@ -27,11 +27,11 @@
 #' names) if the subset of data has less than N rows.
 #'
 #' @export
-subset_data <- function(data, subset_rules) {
+subset_data <- function(data, subset_rules, threshold = 5L) {
 
   if (is.null(subset_rules)) {
     print("No subset rules are given.")
-    return(result)
+    return(data)
```
Comment (frankcorneliusmartin) [inline vtg.preprocessing/R/subset_data.R:34]: My god... That's why kaplan meier did not work without `subset_rules` DOH.

File: vtg.summary/src/R/RPC_summary.R [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+#' @param types types of the columns
+#' @param subset_rules Rules to filter data with. Default is NULL.
+#' @param is_extend_data Whether to extend the data with the `is_extend_data`
+#' function. Default is TRUE.
+#'
+#' @return A list with the following results for each column: length, sum,
+#' range and number of rows with non-empty and empty values.
+#'
+#' @TODO check if works with single column
+RPC_summary <- function(data, columns, threshold = 5L, types = NULL,
```
Comment (frankcorneliusmartin) [inline vtg.summary/src/R/RPC_summary.R]: This `threshold` cannot be user defined.. Read it of a custom environment variable: `SUBSET_MINIMAL_NUM_OBSERVATIONS` or something similar

Comment (frankcorneliusmartin) [reply vtg.summary/src/R/RPC_summary.R]: Same goes for all other `RPC_` functions

File: vtg.summary/src/Makefile
Comment (frankcorneliusmartin) [inline vtg.summary/src/Makefile:1]: This file can be deleted

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]
Can you make these small changes?

File: vtg.summary/src/Makefile
Comment (bartvanb) [reply vtg.summary/src/Makefile:1]: Done

File: Makefile (hunk 1 above)
Comment (bartvanb) [reply Makefile:16]: Not necessary as `IMAGE=starter/${PKG_NAME}` so already dependent on that

**Review by frankcorneliusmartin — APPROVED** [review]

**Pull request merged**